README.md

```diff
@@ -1,12 +1,114 @@
 dp-elasticsearch
 ================
 
-Elasticsearch library to create an elasticsearch client to be able to make requests to elasticsearch.
+Elasticsearch library to create an elasticsearch client to be able to make requests to elasticsearch.Currently the library support 2 versions of elasticsearch 2.2 and 7.10. The version 7.10 uses go-elasticsearch library behind the scenes and this library can be viewed as a wrapper around go-elasticsearch library.  Please follow readme on how to create different versions of client and how to consume these.  
 
 ### elasticsearch package
 
 Includes implementation of a health checker, that reuses the elasticsearch client to check requests can be made against elasticsearch cluster and known indexes.
 
+### setup elasticsearch client
+
+#### setup ES 2.2 client
+
+```
+import (
+    dpEs "github.com/ONSdigital/dp-elasticsearch/v3"
+)
+
+...
+	esClient, esClientErr := dpEs.NewClient(dpEsClient.Config{
+		Address:   cfg.esURL,
+	})
+	if esClientErr != nil {
+		log.Fatal(ctx, "Failed to create dp-elasticsearch client", esClientErr)
+	}
+...
+```
+#### setup ES 7.10 client
+
+Setting up ES7.10 client is similar as setting up es2.2, just that we need to specify client library as ```GoElasticV710```, as follows:
+```
+import (
+    dpEs "github.com/ONSdigital/dp-elasticsearch/v3"
+)
+
+...
+	esClient, esClientErr := dpEs.NewClient(dpEsClient.Config{
+		ClientLib: dpEsClient.GoElasticV710,
+		Address:   cfg.esURL,
+	})
+	if esClientErr != nil {
+		log.Fatal(ctx, "Failed to create dp-elasticsearch client", esClientErr)
+	}
+...
+```
+
+###### Embedding custom http roundtripper with es7.10
+
+You could create custom roundtripper (say if you have to sign requests if you are using es7.10), as follows:
+```
+import (
+    dpEs "github.com/ONSdigital/dp-elasticsearch/v3"
+)
+
+...
+if cfg.signRequests {
+		fmt.Println("Use a signing roundtripper client")
+		awsSignerRT, err := awsauth.NewAWSSignerRoundTripper(cfg.aws.filename, cfg.aws.filename, cfg.aws.region, cfg.aws.service,
```

Comment:
You probably want to describe the awsauth package; if I didn't know this existed in dp-net I would struggle to get this working

Reply:
Again one would need to import cfg, better to use placeholder names for filename, region, service etc.

You could also refer to the dp-net documentation here, in case we change how to setup a roundtripper: https://github.com/ONSdigital/dp-net/tree/main/awsauth#round-tripper

Reply:
fixed in update 1.